pandas/_libs/tslibs/strptime.pyx

```diff
@@ -60,15 +64,37 @@ cdef bint format_is_iso(f: str):
     Generally of form YYYY-MM-DDTHH:MM:SS - date separator can be different
     but must be consistent.  Leading 0s in dates and times are optional.
     """
-    excluded_formats = ["%Y%m"]
-
-    for date_sep in [" ", "/", "\\", "-", ".", ""]:
-        for time_sep in [" ", "T"]:
-            for micro_or_tz in ["", "%z", ".%f", ".%f%z"]:
-                iso_fmt = f"%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S{micro_or_tz}"
-                if iso_fmt.startswith(f) and f not in excluded_formats:
-                    return True
-    return False
+    iso_regex = re.compile(
+        r"""
+        ^                      # start of string
+        (?:                    # group for alternative date formats
+        %Y[-]%m[-]%d           # year month & date separated by either hyphen
+        | %Y[/]%m[/]%d         # OR separated by forward slash
+        | %Y[.]%m[.]%d         # OR separated by period
+        | %Y[ ]%m[ ]%d         # OR separated by space
+        | %Y[\\]%m[\\]%d       # OR separated by backslash
+        | %Y%m%d               # OR no separators
```

Comment:
this is getting too long - can you use a backreference to match the first captured separator? https://www.pythontutorial.net/python-regex/python-regex-backreferences/

Reply:
@MarcoGorelli what about this 
```python 
%Y([-/.\\ ])%m\1%d  # year month & date separated by either ([-/.\\ ])
```

Reply:
maybe, try and see

Reply:
That worked